resources/sqlite/src/conn_opts.rs

```diff
@@ -0,0 +1,263 @@
+/// file-based database named `default_db.sqlite` with `create_if_missing == true`. Use the `filename` and/or
+/// `in_memory` methods to configure the type of database created.
+///
+/// Note that Shuttle does currently not support the `collation`, `thread_name`, `log_settings`, `pragma`, `extension`
+/// options.
+#[derive(Deserialize, Serialize)]
+pub struct SQLiteConnOpts {
+    // The full connection string we construct internally in `ResourceBuilder::output`.
+    pub(crate) conn_str: String,
+    // Mirrored options from the original.
+    pub(crate) filename: Cow<'static, Path>,
+    pub(crate) in_memory: bool,
+    pub(crate) read_only: bool,
+    pub(crate) create_if_missing: bool,
+    pub(crate) shared_cache: bool,
+    pub(crate) statement_cache_capacity: usize,
+    pub(crate) busy_timeout: Duration,
+    pub(crate) immutable: bool,
+    pub(crate) vfs: Option<Cow<'static, str>>,
+
+    pub(crate) pragmas: IndexMap<Cow<'static, str>, Option<Cow<'static, str>>>,
+
+    pub(crate) command_channel_size: usize,
+    pub(crate) row_channel_size: usize,
+
+    pub(crate) serialized: bool,
+}
+
+impl Default for SQLiteConnOpts {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl SQLiteConnOpts {
+    pub fn new() -> Self {
+        let mut pragmas: IndexMap<Cow<'static, str>, Option<Cow<'static, str>>> = IndexMap::new();
+
+        pragmas.insert("key".into(), None);
+
+        pragmas.insert("cipher_plaintext_header_size".into(), None);
+
+        pragmas.insert("cipher_salt".into(), None);
+
+        pragmas.insert("kdf_iter".into(), None);
+
+        pragmas.insert("cipher_kdf_algorithm".into(), None);
+
+        pragmas.insert("cipher_use_hmac".into(), None);
+
+        pragmas.insert("cipher_compatibility".into(), None);
+
+        pragmas.insert("cipher_page_size".into(), None);
+
+        pragmas.insert("cipher_hmac_algorithm".into(), None);
+
+        pragmas.insert("page_size".into(), None);
+
+        pragmas.insert("locking_mode".into(), None);
+
+        pragmas.insert("journal_mode".into(), None);
+
+        pragmas.insert("foreign_keys".into(), Some("ON".into()));
+
+        pragmas.insert("synchronous".into(), None);
+
+        pragmas.insert("auto_vacuum".into(), None);
+
+        Self {
+            conn_str: String::new(),
+            filename: Cow::Borrowed(Path::new(":memory:")),
+            in_memory: false,
+            read_only: false,
+            // Different to what sqlx does.
+            create_if_missing: true,
+            shared_cache: false,
+            statement_cache_capacity: 100,
+            busy_timeout: Duration::from_secs(5),
+            immutable: false,
+            vfs: None,
+            pragmas,
+            serialized: false,
+            command_channel_size: 50,
+            row_channel_size: 50,
+        }
+    }
+
+    /// Use this to set the database to in memory mode.
+    pub fn in_memory(mut self, on: bool) -> Self {
+        self.in_memory = on;
+        self
```

Comment:
This being public does not make sense if a user were to turn this off but still have the default :memory: file. I think it can use in memory by default, and only calls to filename() will flip this flag and set the new filename.

Reply:
We don't want to give the user access to the connection string directly because for security reasons (user could construct arbitrary paths). That's why we're constructing the conn string internally and need to expose this is some other way.

Reply:
Is path injection possible with the current impl? i.e. `filename = "../../../some/path"`
Btw, is that what you mean by arbitrary paths, or is there other behaviors to avoid?